src/main/java/org/spongepowered/common/registry/type/text/TextColorRegistryModule.java

```diff
@@ -60,20 +60,20 @@
     @Override
     public void registerDefaults() {
         addTextColor(TextFormatting.BLACK, Color.BLACK);
-        addTextColor(TextFormatting.DARK_BLUE, Color.ofRgb(0x0000AA));
-        addTextColor(TextFormatting.DARK_GREEN, Color.ofRgb(0x00AA00));
-        addTextColor(TextFormatting.DARK_AQUA, Color.ofRgb(0x00AAAA));
-        addTextColor(TextFormatting.DARK_RED, Color.ofRgb(0xAA0000));
-        addTextColor(TextFormatting.DARK_PURPLE, Color.ofRgb(0xAA00AA));
-        addTextColor(TextFormatting.GOLD, Color.ofRgb(0xFFAA00));
-        addTextColor(TextFormatting.GRAY, Color.ofRgb(0xAAAAAA));
-        addTextColor(TextFormatting.DARK_GRAY, Color.ofRgb(0x555555));
-        addTextColor(TextFormatting.BLUE, Color.ofRgb(0x5555FF));
-        addTextColor(TextFormatting.GREEN, Color.ofRgb(0x55FF55));
-        addTextColor(TextFormatting.AQUA, Color.ofRgb(0x55FFFF));
-        addTextColor(TextFormatting.RED, Color.ofRgb(0xFF5555));
-        addTextColor(TextFormatting.LIGHT_PURPLE, Color.ofRgb(0xFF55FF));
-        addTextColor(TextFormatting.YELLOW, Color.ofRgb(0xFFFF55));
+        addTextColor(TextFormatting.DARK_BLUE, Color.of(0x0000AA));
+        addTextColor(TextFormatting.DARK_GREEN, Color.of(0x00AA00));
+        addTextColor(TextFormatting.DARK_AQUA, Color.of(0x00AAAA));
+        addTextColor(TextFormatting.DARK_RED, Color.of(0xAA0000));
+        addTextColor(TextFormatting.DARK_PURPLE, Color.of(0xAA00AA));
+        addTextColor(TextFormatting.GOLD, Color.of(0xFFAA00));
```

Comment:
The more I look at this, the more I prefer the `ofRgb` instead of just flag `of`, since you can define how it's formatted.

Reply:
I'm fine with that; though in my mind it's more consistent with the naming of other static methods used throughout Sponge. If we do revert this change, then I would like to be consistent and use `ofVector3i`, `ofJavaColor`, and so on.

Reply:
`ofVector3i` should also be `ofRgb` as its simply an overload to take a vector.

Reply:
Its ensuring that we don't back ourselves into a confusion corner later if say we wanted to add an `ofHSV` (not a bad idea as it is a much easier color space to work with for many use cases).

Reply:
We currently store colors in an RGB format, meaning that we would either need to have a storage option for HSV (preferable) or handle the conversion, which is messy and limits our ability to have getters for HSV values. In a graphical environment, you would have an `RgbColor` class and a `HsvColor` class and use inheritance to handle rendering them. I don't believe this step is at all necessary or within the scope of Sponge. Colors in Minecraft are handled with RGB values, and I strongly recommend to keep this trend as there is not sufficient reason to go to these lengths to support HSV or any other color format.

@gabizou, thoughts on this and the resolution of the `of`/`ofRgb` change?

Reply:
I think you're overcomplicating and thus overthinking what is being asked here. We're not asking you to implement any of this, we're asking for a function name change to try to futureproof the API. I, for one, am in favour of `ofRgb` instead of `of`.

The process of devloping an API is not to think about what is now, but also about what could be in the future. When I suggested we should give this the whole API/impl split treatment, I was after the current functionality being split to reduce the API footprint to make it easier to add future API methods later and give us flexibility in the implementation. `ofHSV` is a potential extension _that can be done later_. Futher, the whole point of splitting this into API/impl is that we can do that sort of subclassing if we need to without impacting the API. By naming it `ofRgb` now, we are ensuring that we don't need to break backwards compatibility because we have a method that is already describing what is going on. If we leave it as `of`, and then want to add `ofHSV`, we then end up with inconsistent naming which would likely lead to the deprecation of `of` and adding `ofRgb` anyway. `ofRgb` is also more descriptive to just look at.

No-one is saying that we should stop storing values as RGB, and these are potentially valid and interesting questions, but these are implementation details that can be picked up later. We want to design the API such that any implementation detail, be it storage detail or otherwise, can fit the API nicely without needing any more breakages.

Reply:
My point with this is that it *would* require breakages. The `Color` interface defines `getRed`, `getGreen`, `getBlue`, and other methods related to the RGB format. If there is a desire to pursue other formats like HSV in the future, then we need to consider that now and redesign the API to support that.

If the primary argument for `ofRgb` over `of` relies on future implementation that the API doesn't support, then for it to be valid we need to update the API to support multiple color formats now.

Reply:
An `ofHSV` function which is a simple conversion into the RGB color space would be simple and non-breaking. No need to change the backing storage, or even consider that right now. We just don't want a plain `of` function that doesn't even make clear in the simplest case between `ofRGB` and `ofBGR`.

Reply:
Converting HSV to RGB is not something I consider worthwhile (see [this question](https://stackoverflow.com/questions/3018313/algorithm-to-convert-rgb-to-hsv-and-hsv-to-rgb-in-range-0-255-for-both) that has the conversion formulas). For reference, the `java.util.Color` class is defined to represent colors in [sRGB](http://www.w3.org/pub/WWW/Graphics/Color/sRGB.html) format (link from the class javadocs). If a developer would prefer to use HSV, they can use the conversion methods in `java.awt.Color`.

`Color.of(int rgb)` provides the same specification that it is in RGB format as does `new java.awt.Color(int rgb)`. I maintain that the `Color` class should be defined to use the RGB format and use `#of`, as `#ofRgb` is unnecessarily repetitive and is different from the `#of` standard used elsewhere in Sponge.

Reply:
I think I’ve said this before, the method should very well be named ofRgb to avoid confusion in the format the colors are presented. Java API is not a standard to hold us up to on making such a method name decision. I’m putting my decision as it was mentioned before.

Reply:
Here's what I'm tracking so far. What are we doing with the one for `java.awt.Color`?

`#of(int rgb)` -> `#ofRgb(int rgb)`
`#of(int red, int green, int blue)` -> `#ofRgb(int red, int green, int blue)`
`#of(Vector3i vector3i)` -> `#ofRgb(Vector3i vector3i)`
`#of(java.awt.Color)` -> ??? (`#ofJavaColor`?)

Reply:
Can make it ofColor since the class type is implied by the argument type. Otherwise fine.